### DISCUSSION TITLE
Compile time options for when the release binary is signed

### DISCUSSION THREAD

**Discussion by habys** [ipxe/ipxe#1236] [General]
Hello, I am building my own IPXE binary currently but would like to use the signed binary when that happens. Currently I need to enable the DOWNLOAD_PROTO_HTTPS option in the source as I am using HTTPS. Once the release binary is signed, I won't be able to build my own and still have secure boot enabled with the default key anymore right?

If this understanding is correct, is there any way to eventually get the signed IPXE binary that supports HTTPS?

**Comment by NiKiZe** [comment]
This has been discussed before but unable to find that conversation right now.
The request was for having `DOWNLOAD_PROTO_HTTPS` enabled by default in efi builds.

Unfortunately there is instances where the system is not good at creating entropy, which essentially makes iPXE feel hanged, or takes a long time to start on those systems.
With that said, it wouldn't be unlikely that signed builds would exist with https enabled as well.

**Reply by habys** [reply]
That would be fantastic. I have no great concern if there is a delay to collect entropy, but I don't know what the path forwards will be if there is not a signed binary that supports HTTPS. We've been manually disabling secure boot on all of our machines and would like to stop.

Thanks for the response.
